### ISSUE TITLE
Delete Published Resources. How do you remove a published (DOI) resource?

### ISSUE THREAD

**Issue by ChristinaB** [hydroshare/hydroshare#1607]
I need help deleting these two published resources. 
https://www.hydroshare.org/resource/726781eb2bbc4839ad56eed73ba67e85/
 http://www.hydroshare.org/resource/61dbf283189c457185ea16bc629d9db4/

New user Sarah Jenkins had the settings as private. She thought the publish button was a required last step to push the data to the server when Creating a new resource.  She did not realize that is was generating a DOI and going public.  
This particular resource needs to be PRIVATE for publication issues, this was our use case dataset to highlight how useful hydroshare would be to keep it private (oh boy).  We will recreate the resource properly. 

Either we need training before throwing in new users to the platform, or to keep working on more active messaging that helps new users more.  

@rayi113 Can you help me and quickly delete?
@dtarb What is the design plan for deleting published resources? Here was the workflow,
1. Create New; ESRI Shapefile
2. Wait for a long time for a big file to upload
3. Hit Edit pencil.  Save each box.  Keep on Private. Shareable.
4. Save each edit. 
5.  THe message 'Sufficient to make public or publish' was confusing.  She wanted to publish it to the group of people with sharing settings.
6. She was waiting for a 'I'm done' message which doesn't come. 
7. She went back to My resources icon (not sharing)(
8. She saw the globe option and assumed that was the last step to preserve it permanently on the server.    She sees the Publication Agreement and assumed that was part of the ordinary user requirements to upload a resource to HydroShare.    She got the DOI.
9.  She couldn't delete.  She saw a message that 'changes can only be made to a new version of the resources' and that it would be overwritten.  The process stored two copies, doubling the issue, rather than overwriting the previous version as expected.  
2. The Help file, Creating Resources,

**Comment by horsburgh** [comment]
Looks like we need to improve our messaging. This isn't the first time this week that a user has published something not knowing that it is a permanent action.

We need to be clear when we are talking to people that "share" and "publish" are not the same. Nor are "public" and "published" the same thing.  I suspect that we need a dialog box on the publish step that says something like the following:

"Are you sure you want to publish this resource? You will NOT be able to edit your resource or its metadata once it is published. This is a permanent step and should only be done when you are finished modifying your resource and you are ready to formally publish it. You can make your resource public or share with specific HydroShare users without formally publishing it."

**Comment by dtarb** [comment]
I suggest we institute a procedure where when a user clicks publish we
1.  Ask them whether they are sure they want to publish it using the wording Jeff wrote above.
2.  Indicate that it will sit as publication pending for a week, and then the DOI will be registered.
3.  Email all parties (authors, contributors, owners) for which HydroShare has email that this resource is being published.
4.  Have a button/link where they can withdraw publication during the pending period and have this link be in the email that is sent out as well as on the resource landing page.
5.  We refine the wording to be clear to try avoid these problems.  Could Jeff and Christina please come up with refined wording that communicates the process clearly.  I think we need a help page that explains in more detail how things work with links from the publish dialog to "learn more" that go to this.

**Comment by hyi** [comment]
@ChristinaB I have removed published status for these two resources and changed them to be private for now. You can ask the author Sarah Jenkins to check them and do whatever she needs with these two resources now.

**Comment by horsburgh** [comment]
@dtarb - I created a new issue last week for the revised data publication workflow. See #1608. I suggest we address your 5 steps there. Can you review what I put in #1608 and make sure it covers what we discussed?

@ChristinaB - since these resources have been published, and we are now "unpublishing" them, they need to be deleted and re-created if you ever want to publish them again. The process of "unregistering" the DOIs will reserve them from ever being registered again, and so if you keep these resources with the same HydroShare identifiers, they will not be able to be published with a DOI since the DOI is based on the HydroShare identifier.

**Comment by dtarb** [comment]
The root cause of this problem has been fixed with #1607. 

The documentation/policy aspect of this is that a published resource cannot be deleted by the user and is only deleted by the system administrator in exceptional cases (e.g. grave error or abuse).  Please make sure that the help files reflect this.  Refer also to #1607 and #1649 that indicates an additional safeguard that will be implemented.

What remains is to sort out the two specific resources mentioned that the user Sarah Jenkins created.  @ChristinaB please work with her, since you guided her in creating these resources.  They need to be deleted after she has saved any files in them that she wants to keep.  We are interpreting this as a case of grave error and @hyi has already as an administrator changed the status from published to private.   Once these are deleted we will ask @Lizabrazil to unregister the DOI's with crossref.  

If the user Sarah Jenkins wants these files to be in HydroShare she should re-upload them into a new resource with different GUID and delete the old one.  

@ChristinaB I am assigning back to you.  When the resources are deleted, please alert and assign to @Lizabrazil for the DOI unregistering.  Then when that is done this issue can be closed.

**Comment by ChristinaB** [comment]
@hyi  - can you delete this resource so that this link is broken forever?  
Jenkins, S. (2016). Water Point Data Exchange global dataset failures, joined to original data, HydroShare, http://dx.doi.org/10.4211/hs.61dbf283189c457185ea16bc629d9db4

@Lizabrazil can you delete this DOI?  

We have a copy of the correct version of how we should have created this resource - without publishing the DOI!

**Comment by hyi** [comment]
@ChristinaB Have just deleted this resource along with its wrongly created versioned resource counterpart. Let me know if you notice any issues.

**Comment by dtarb** [comment]
Received email from @Lizabrazil indicating that Crossref is unable to unregister the DOI's as they have already been indexed.  The DOI's point to:
https://www.hydroshare.org/resource/726781eb2bbc4839ad56eed73ba67e85/
https://www.hydroshare.org/resource/61dbf283189c457185ea16bc629d9db4/
which are HydroShare URL's that do not exist.

Options for addressing this include
A. Establish pages at the above URL's that state: "This resource has been removed at the request of its authors."
B. Establish a generic page that states: "This resource has been removed at the request of its authors." and set redirects from the above URL's to this within HydroShare.
C. Establish a generic page within HydroShare that states: "This resource has been removed at the request of its authors." and request that Crossref change its registration to point to this generic page.
D. Establish a generic page outside HydroShare (e.g. on CUAHSI web page) that states: "This resource has been removed at the request of its authors." and request that Crossref change its registration to point to this generic page.
E. Do nothing and leave these as broken links.

I set out to attempt A using the admin interface but practicing on beta.hydroshare.org to work out the procedure first.  I encountered a problem that prevents listing pages/resources in the admin interface, #1856.

**Comment by hyi** [comment]
@dtarb Options C and D involve crossref or CUAHSI, and options A and B can be done within HydroShare. Between options A and B, I prefer B over A. In fact, I am trying to see if I can retrieve the original referrer info from the redirect request from crossref to hydroshare landing page, if I can do that, I may just customize a 404 page to handle this special case, but so far, Django does not seem to show my customized 404 page for some reason. Will try to see if I can get customized 404 page to work to handle this special case first. If I cannot get it to work, I can try option B for redirect.

**Comment by Lizabrazil** [comment]
Hi,

Crossref support originally suggested option C. I'm not sure how hydroshare creates DOIs on the back end. For CUAHSI, we submit an XML that indicates the URL we want the DOI to point to. Whenever a change is needed, we can submit a new XML that includes the same DOI with a new URL.

**Comment by dtarb** [comment]
Let's go with option C then. @hyi lets discuss on call.  @Lizabrazil we'll send you the URL to the page when it is set up.

**Comment by hyi** [comment]
@dtarb @Lizabrazil Option C is the easiest on HydroShare side, which means I don't have to customize the standard 404 page, but only create a page for crossref to redirect to.

**Comment by dtarb** [comment]
The page https://www.hydroshare.org/removed/ has been created.  @Lizabrazil please change crossref for these resources to point to this page.

**Comment by ChristinaB** [comment]
@dtarb 
At UCGIS and other events, we have a 5-10% rate of new users published to a DOI thinking that is what 'publishes' data on HydroShare.  Is this issue ready to be closed or should we continue working on it?

**Comment by ChristinaB** [comment]
@hyi -- do you think this issue is ready to close?   We would like to still address the 5-10% accidental DOI problem, but we could start a new issue to do that.

**Comment by hyi** [comment]
@ChristinaB For now, wrongly published resources can be retracted/deleted by manual process, so I'd suggest to close this issue and start a new issue if we want to address accidental DOI publishing problem systematically.

**Comment by dtarb** [comment]
Closing as this is superceded by manual review process outlined in #2298 and CUAHSI policy decisions for a manual review process.

The only way to delete a published resource now, and once implementation of #2298 is complete will be to request from CUAHSI system administrators that it be deleted.
